common-controller/internal/operator/constant/constant.go

```diff
@@ -23,6 +23,7 @@ const (
 	ApplicationController        string = "ApplicationController"
```

Comment:
Let's add a go test for this. Basically we can deploy the gatewayclass and verify the status.

In the Gateway Kind, gatewayclassname is a mandatory field. So we should link these two CRs.

Reply:
Added integration test